src/app/auth.guard.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { Router } from '@angular/router';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+  constructor(private router: Router) { }
```

Comment:
Inject here `AuthenticationService`, and then in the `canActivate` method perhaps try doing
```
let isLoggedIn = this.authenticationService.isAuthenticated();
if (isLoggedIn){
  return true
} else {
  this.router.navigate(['/dashboard']);
}
```